### PR TITLE
Port implementation of fast decimal formatter from monolithic app

### PR DESCRIPTION
**Pull request by mqchau**
We have this function in our monolithic app. Now another app is looking to use it so I refactor it into a standalone gem and make it open-source.

### CODE REVIEW

File: spec/ae_fast_decimal_formatter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+require 'spec_helper'
+
+describe 'AeFastDecimalFormatter' do
+  def ae_fast_decimal_formatter(number, precision)
+    AeFastDecimalFormatter.new(number, precision).format
```
Comment (wpliao1989) [inline spec/ae_fast_decimal_formatter_spec.rb:5]: Is it possible to make the interface a class method?
Like `AeFastDecimalFormatter.format(number, precision)` to be consistent with what we currently have and it feels redundant to first initialize an object and call `format`.

Comment (mqchau) [reply spec/ae_fast_decimal_formatter_spec.rb:5]: One thing I remember from Sandi Metz class is that class method is less flexible to extend than instance method.

File: ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  }
+  return rb_str_new2(str);
+}
+
+struct ae_fast_decimal_formatter {
+  double num;
+  int precision;
+};
+
+static void ae_fast_decimal_formatter_free(void *p) {}
```
Comment (wpliao1989) [inline ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c:78]: Do we need to free the internal pointer?

Comment (wpliao1989) [reply ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c:78]: https://github.com/rubygems/guides/blob/173db855aeb25193616cc0f8eed2928f8bfa626d/ext/my_malloc/my_malloc.c#L22

Comment (mqchau) [reply ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c:78]: No need to free. We didn't malloc anything. This object in in stack, not in heap.

**Review by dvicory — APPROVED** [review]
I wouldn't take my review as the golden seal of approval, since I have no prior experience with native Ruby extensions, but everything generally looks in order. Really glad to see that benchmark test there too.

File: ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c (hunk 2 above)
Comment (wpliao1989) [reply ext/ae_fast_decimal_formatter/ae_fast_decimal_formatter.c:78]: Cool that makes sense.

**Review by wpliao1989 — APPROVED** [review]

**Pull request merged**